sql/descriptor.go

```diff
@@ -62,12 +62,15 @@ type descriptorProto interface {
 }
 
 // checkPrivilege verifies that p.user has `privilege` on `descriptor`.
-func (p *planner) checkPrivilege(descriptor descriptorProto, privilege privilege.Kind) error {
-	if descriptor.GetPrivileges().CheckPrivilege(p.user, privilege) {
+func (p *planner) checkPrivilege(descriptor descriptorProto, priv privilege.Kind) error {
+	if descriptor.GetPrivileges().CheckPrivilege(p.user, priv) {
 		return nil
 	}
+	if priv != privilege.SELECT && isSystemConfigID(descriptor.GetID()) {
+		return fmt.Errorf("system database is read-only")
```

Comment:
Since it's more nuanced than I'd originally thought, it might not be worth the effort to improve this error for all cases. I could eventually cover this in the docs somewhere?